### DISCUSSION TITLE
Consider using smarter "fetch" on node.js

### DISCUSSION THREAD

**Discussion by kigawas** [wevm/viem#1413] [Idea / Feature Request]
Currently due to the usage of [experimental global fetch API](https://github.com/wagmi-dev/viem/blob/main/src/utils/rpc.ts#L94), viem's transport does not work behind proxy. 

Reference: https://github.com/nodejs/node/issues/42814

By the way the polyfilled `fetch` is [undici](https://github.com/nodejs/undici) which can also be disabled by `NODE_OPTIONS='--no-experimental-fetch'`

**Comment by jxom** [comment]
Wonder if we expose an API so you can provide your own fetch implementation. Thoughts?

**Reply by kigawas** [reply]
Maybe `fetch` can also be an argument

```ts
async function http<TBody extends RpcRequest | RpcRequest[]>(
  url: string,
  { body, fetchOptions = {}, timeout = 10_000 }: HttpOptions<TBody>,
): Promise<HttpReturnType<TBody>> {
  const { headers, method, signal: signal_ } = fetchOptions
  try {
    const response = await withTimeout(
      async ({ signal }) => {
        const response = await fetch(url, {
          ...fetchOptions,
```
